File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[Arrays.stream(arr).sum()];
        int start = 0;
        
        for(int i = 0; i < arr.length; i++){
            if(arr[i] != 1) {
                Arrays.fill(answer, start, start + arr[i], arr[i]);
            }else {
                answer[start] = arr[i];
            }
            start += arr[i];
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len) {
    int len = 0;
    int idx = 0;
    
    for(int i = 0; i < arr_len; i++){
        len += arr[i];
    }
    
    int* answer = (int*)malloc(len * sizeof(int));
    
    for(int j = 0; j < arr_len; j++){
        int n = arr[j];
        for(int k = 0; k < n; k++){
            answer[k + idx] = n;
        }
        idx += n;
    }
    
    return answer;
}